### PR TITLE
MGMT-7568: ocp_version_update - support os/release images

### PR DESCRIPTION
**Pull request by danielerez**
# Assisted Pull Request

## Description

- In order to ensure correlation between the default versions jsons, update also default_os_images.json and default_release_images.json.
- As release and rhcos fields are optional in default_ocp_versions.json, update them only if available.
I.e. these fields are just a fallback in case that os/release images files are missing.

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

Tested using a 'dry-run', still need to test the full flow in jenkins. 

## Assignees

/cc @oshercc 
/cc @gamli75 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: tools/ocp_version_update/update_default_release_versions_to_latest.py [hunk 1]
```diff
@@ -231,11 +237,13 @@ def get_release_note_url(minor_release: str):
     return release_url
 
 
-def get_release_data(minor_release):
-    release_data = subprocess.check_output(OCP_INFO_CALL.format(version=minor_release), shell=True)
+def get_release_data(minor_release, cpu_architecture):
+    if cpu_architecture == "x86_64":
+        cpu_architecture = "amd64"
```
Comment (gamli75) [inline tools/ocp_version_update/update_default_release_versions_to_latest.py:242]: why we are replacing the cpu_architecture? please add a remark

File: tools/ocp_version_update/update_default_release_versions_to_latest.py [hunk 2]
```diff
@@ -333,50 +367,129 @@ def main(args):
...
+
+    if dry_run:
+        jira_client, task = None, "TEST-8888"
+    else:
+        jira_client, task = create_task(args, TICKET_DESCRIPTION + " " + versions_str)
+
+    title = PR_MESSAGE.format(task=task, versions_string=versions_str)
+    logger.info(f"PR title will be {title}")
+
+    #user, password = get_login(args.github_user_password)
```
Comment (gamli75) [inline tools/ocp_version_update/update_default_release_versions_to_latest.py:474]: please remove it if not needed

**Review by gamli75 — COMMENTED** [review]
How did you test it? I think that we must add some tests for this tool

File: tools/ocp_version_update/update_default_release_versions_to_latest.py (hunk 2 above)
Comment (danielerez) [reply tools/ocp_version_update/update_default_release_versions_to_latest.py:474]: just for testing... uncommented.

File: tools/ocp_version_update/update_default_release_versions_to_latest.py (hunk 1 above)
Comment (danielerez) [reply tools/ocp_version_update/update_default_release_versions_to_latest.py:244]: sure, added a comment

File: tools/ocp_version_update/update_default_release_versions_to_latest.py [hunk 3]
```diff
@@ -123,11 +129,11 @@ def create_task(args, description: str):
     return jira_client, task
 
 
-def get_default_release_json():
-    res = requests.get(ASSISTED_SERVICE_MASTER_DEFAULT_OCP_VERSIONS_JSON_URL)
+def get_default_release_json(json_url):
```
Comment (YuviGold) [inline tools/ocp_version_update/update_default_release_versions_to_latest.py:132]:
```suggestion
def request_json_file(json_url):
```

File: tools/ocp_version_update/update_default_release_versions_to_latest.py [hunk 4]
```diff
@@ -48,6 +50,10 @@
 ASSISTED_SERVICE_UPSTREAM_URL = f"https://github.com/{ASSISTED_SERVICE_GITHUB_REPO}.git"
 ASSISTED_SERVICE_MASTER_DEFAULT_OCP_VERSIONS_JSON_URL = \
     f"https://raw.githubusercontent.com/{ASSISTED_SERVICE_GITHUB_REPO}/master/data/default_ocp_versions.json"
+ASSISTED_SERVICE_MASTER_DEFAULT_OS_IMAGES_JSON_URL = \
+    f"https://raw.githubusercontent.com/{ASSISTED_SERVICE_GITHUB_REPO}/master/data/default_os_images.json"
+ASSISTED_SERVICE_MASTER_DEFAULT_RELEASE_IMAGES_JSON_URL = \
+    f"https://raw.githubusercontent.com/{ASSISTED_SERVICE_GITHUB_REPO}/master/data/default_release_images.json"
```
Comment (YuviGold) [inline tools/ocp_version_update/update_default_release_versions_to_latest.py:56]: Have the prefix as a constant and compound the full path using `DEFAULT_VERSIONS_FILES`, `DEFAULT_OS_IMAGES_FILE`, and `DEFAULT_RELEASE_IMAGES_FILE`

File: tools/ocp_version_update/update_default_release_versions_to_latest.py [hunk 5]
```diff
@@ -292,21 +301,37 @@ def main(args):
...
+    updated_version_json = copy.deepcopy(default_version_json)
+
     for release in default_version_json:
 
         if release in SKIPPED_MAJOR_RELEASE:
             logger.info(f"Skipping {release} listed in the skip list")
             continue
 
-        latest_ocp_release = get_latest_release_from_minor(release)
+        latest_ocp_release = get_latest_release_from_minor(release, "amd64")
```
Comment (YuviGold) [inline tools/ocp_version_update/update_default_release_versions_to_latest.py:334]: const

File: tools/ocp_version_update/update_default_release_versions_to_latest.py [hunk 6]
```diff
@@ -333,50 +369,129 @@ def main(args):
...
+
+def create_jira_task(updates_made_str, dry_run, args):
+    logger.info(f"changes were made on the following versions: {updates_made_str}")
+
+    versions_str = ", ".join(updates_made_str)
+
+    if dry_run:
+        jira_client, task = None, "TEST-8888"
+    else:
+        jira_client, task = create_task(args, TICKET_DESCRIPTION + " " + versions_str)
```
Comment (YuviGold) [inline tools/ocp_version_update/update_default_release_versions_to_latest.py:471]: `jira_client` is never used.

```suggestion
        _, task = None, "TEST-8888"
    else:
        _, task = create_task(args, TICKET_DESCRIPTION + " " + versions_str)
```

Comment (danielerez) [reply tools/ocp_version_update/update_default_release_versions_to_latest.py]: right, done

File: tools/ocp_version_update/update_default_release_versions_to_latest.py (hunk 5 above)
Comment (danielerez) [reply tools/ocp_version_update/update_default_release_versions_to_latest.py]: done

File: tools/ocp_version_update/update_default_release_versions_to_latest.py (hunk 4 above)
Comment (danielerez) [reply tools/ocp_version_update/update_default_release_versions_to_latest.py]: done

File: tools/ocp_version_update/update_default_release_versions_to_latest.py (hunk 3 above)
Comment (danielerez) [reply tools/ocp_version_update/update_default_release_versions_to_latest.py]: done

**Review by YuviGold — APPROVED** [review]

**Comment by oshercc** [comment]
/hold
want to test first

**Pull request merged**